[RFR] Update pre-commit hooks rev

digitronik opened the pull request:
<!-- Make sure to prefix the PR Title with any one of the below options...

   [WIP] - Work in Progress
   [WIPTEST] - Work in Progress with PRT Testing
   [RFR] - Ready for 1st Review
   [1LP][WIP] - 1st Level Pass, Work in Progress
   [1LP][WIPTEST] - 1st Level Pass, Work in Progress with PRT Testing
   [1LP][RFR] - 1st Level Pass, Ready for 2nd Review
-->

## Purpose or Intent

- These are new changes related to pyupgrade in fever of PR
https://github.com/ManageIQ/integration_tests/pull/10027

- Mostly related to file read mode as its default.

**NOTE: Changes are related to pre-commit run :D (not maual)**

<!-- Please provide some information related to PR. Some examples are:

- __Fixing__ X because it is currently doing Y which is incorrect. It should be doing Z.
- __Extending__ X because I need it to do Y so that I can update/add more test cases; PR #123
depends on it.
- __Adding tests__ for feature X to cover cases Y and Z. They were implemented in product
version1.2.3.
- __Updating tests__ for feature X because the behavior changed in product version 1.2.3.
- __Enhancement__ for feature X

Note: You can introduce Screenshots/Gifs for providing more information.
-->

### PRT Run
<!--
- It's an internal RH service and only runs against signed whitelisted commits.
- It runs against a single appliance.
- Need to provide specific pytest expression else default it will run smoke tests [-m smoke].
- DOUBLE CURLY BRACES REQUIRED. Examples are in single to not get picked

Some examples are:
- {pytest: cfme/tests/test_foo_file.py -v}
- {pytest: cfme/tests/test_foo_file.py -k "test_foo_1 or test_foo_2" -v}
- {pytest: cfme/tests/ -k "test_foo_1 or test_foo_2 or test_foo_3" -v}
- {pytest: cfme/tests/test_foo_file.py --use-provider rhv43 -v}
- {pytest: cfme/tests/test_foo_file.py --long-running -v}

Notes:
- If PRT fails other than PR purpose/intent; please add a respective comment.
- For any guidance or assistance with PRT results; please contact to maintainers.
-->

dajoRH commented:
I detected some fixture changes in commit c7377a24802be7abb74857ed8a0de90243814275

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`candu_db_restore`** is used in the following files:
- **cfme/tests/candu/test_graph_groupbytag.py**
    - *test_tagwise*

The *global* fixture **`depot_machine_ipv4_and_ipv6`** is used in the following files:
- **cfme/tests/configure/test_log_depot_operation.py**
    - *test_log_collection_over_ipv6*

The *global* fixture **`import_dialog`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
- **cfme/tests/automate/test_service_dialog.py**
    - *custom_button*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_request_details_page_tagcontrol_field*
    - *test_save_dynamic_multi_drop_down_dialog*
- **cfme/tests/services/test_dynamicdd_dialogelement.py**
    - *test_dynamic_submit_cancel_button_service*
- **cfme/tests/services/test_myservice.py**
    - *test_load_service_with_button*
- **cfme/tests/services/test_rest_services.py**
- **cfme/tests/services/test_service_customer_bz.py**
    - *test_edit_import_dialog*

The *local* fixture **`provider_credentials`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**

The *local* fixture **`default_alerts`** is used in the following files:
- **cfme/tests/control/test_default_alerts.py**
    - *test_default_alerts*

The *local* fixture **`vm_ownership`** is used in the following files:
- **cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py**
    - *resource_usage*
    - *chargeback_report_custom*

The *local* fixture **`db_restore`** is used in the following files:
- **cfme/tests/optimize/test_bottlenecks.py**
    - *test_bottlenecks_report_event_groups*
    - *test_bottlenecks_report_show_host_events*
    - *test_bottlenecks_report_time_zone*
    - *test_bottlenecks_summary_event_groups*
    - *test_bottlenecks_summary_show_host_events*
    - *test_bottlenecks_summary_time_zone*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

RonnyPfannschmidt commented on cfme/fixtures/xunit_tools.py:
```diff
@@ -29,7 +29,7 @@
 compiled_blacklist = re.compile('(' + ')|('.join(blacklist) + ')')
 
 
-timestamp = '{:%Y%m%d%H%M%S}'.format(datetime.datetime.now())
```
consider calling the `.strftime` method or a call to the `format` for readability here

digitronik replied:
done

RonnyPfannschmidt commented:
all of those `fauxfactory.gen_$name` that moved into f strings might help readability if `from fauxfactory import gen_$name as faux_name was used`
but that convention also has some downs, it may not be worth it

valaparthvi commented on cfme/fixtures/candu.py:
```diff
@@ -73,7 +73,7 @@ def candu_db_restore(temp_appliance_extended_db):
     db_storage_hostname = conf.cfme_data.bottlenecks.hostname
     db_storage_ssh = SSHClient(hostname=db_storage_hostname, **conf.credentials.bottlenecks)
-    rand_filename = "/tmp/db.backup_{}".format(fauxfactory.gen_alphanumeric())
+    rand_filename = f"/tmp/db.backup_{fauxfactory.gen_alphanumeric()}"
```
```suggestion
rand_filename = fauxfactory.gen_alphanumeric(start="/tmp/db.backup_", length=20)
```

valaparthvi replied:
If you're going to implement this then same change is required everywhere in this PR.

digitronik replied:
@valaparthvi this change more related to pyupgrade. we are adding this in support of last changes. We will do in another PR; I'm avoiding this as inderectly fauxfactory also using `.format`

valaparthvi commented on cfme/tests/cloud/test_quota_tagging.py:
```diff
@@ -166,7 +166,7 @@ def test_quota_tagging_cloud_via_lifecycle(request, appliance, provider, prov_da
     """
     recursive_update(prov_data, {
-        'request': {'email': 'test_{}@example.com'.format(fauxfactory.gen_alphanumeric())}})
+        'request': {'email': f'test_{fauxfactory.gen_alphanumeric()}@example.com'}})
```
Thoughts on using `fauxfactory.gen_email` instead?

valaparthvi commented on cfme/tests/cloud/test_quota_tagging.py:
```diff
@@ -231,7 +231,7 @@ def test_cloud_quota_by_lifecycle(request, appliance, provider, set_entity_quota
     """
     recursive_update(prov_data, {
-        'request': {'email': 'test_{}@example.com'.format(fauxfactory.gen_alphanumeric())}})
+        'request': {'email': f'test_{fauxfactory.gen_alphanumeric()}@example.com'}})
```
`fauxfactory.gen_email()`?

valaparthvi commented on cfme/utils/smem_memory_monitor.py:
```diff
@@ -357,7 +357,7 @@ def run(self):
         except Exception as e:
             logger.error(f'Error in Monitoring Thread: {e}')
-            logger.error('{}'.format(traceback.format_exc()))
+            logger.error(f'{traceback.format_exc()}')
```
```suggestion
logger.error(traceback.format_exc())
```

dajoRH commented:
I detected some fixture changes in commit be481fe15aef0dcf1a5eabaf25d7d5c09db5a00d

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`candu_db_restore`** is used in the following files:
- **cfme/tests/candu/test_graph_groupbytag.py**
    - *test_tagwise*

The *global* fixture **`depot_machine_ipv4_and_ipv6`** is used in the following files:
- **cfme/tests/configure/test_log_depot_operation.py**
    - *test_log_collection_over_ipv6*

The *global* fixture **`import_dialog`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
- **cfme/tests/automate/test_service_dialog.py**
    - *custom_button*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_request_details_page_tagcontrol_field*
    - *test_save_dynamic_multi_drop_down_dialog*
- **cfme/tests/services/test_dynamicdd_dialogelement.py**
    - *test_dynamic_submit_cancel_button_service*
- **cfme/tests/services/test_myservice.py**
    - *test_load_service_with_button*
- **cfme/tests/services/test_rest_services.py**
- **cfme/tests/services/test_service_customer_bz.py**
    - *test_edit_import_dialog*

The *local* fixture **`provider_credentials`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**

The *local* fixture **`default_alerts`** is used in the following files:
- **cfme/tests/control/test_default_alerts.py**
    - *test_default_alerts*

The *local* fixture **`vm_ownership`** is used in the following files:
- **cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py**
    - *resource_usage*
    - *chargeback_report_custom*

The *local* fixture **`db_restore`** is used in the following files:
- **cfme/tests/optimize/test_bottlenecks.py**
    - *test_bottlenecks_report_event_groups*
    - *test_bottlenecks_report_show_host_events*
    - *test_bottlenecks_report_time_zone*
    - *test_bottlenecks_summary_event_groups*
    - *test_bottlenecks_summary_show_host_events*
    - *test_bottlenecks_summary_time_zone*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

digitronik commented:
> all of those `fauxfactory.gen_$name` that moved into f strings might help readability if `from fauxfactory import gen_$name as faux_name was used`
yup

> but that convention also has some downs, it may not be worth it
I added prefix like `start` in fauxfactory to avoid `.format` (mainly py2.7) support :D. But now we moved to py3.7 so that f-string doing this work.

I agreed with you... need to think about more readability.

ganeshhubale commented on cfme/fixtures/candu.py:
```diff
@@ -73,7 +73,7 @@ def candu_db_restore(temp_appliance_extended_db):
+    rand_filename = f"/tmp/db.backup_{fauxfactory.gen_alphanumeric()}"
     db_storage_ssh.get_file("{}/candu.db.backup".format(
         conf.cfme_data.bottlenecks.backup_path), rand_filename)
     app.ssh_client.put_file(rand_filename, "/tmp/evm_db.backup")
```
Are we missing this?

ganeshhubale commented on cfme/tests/optimize/test_bottlenecks.py:
```diff
@@ -39,7 +39,7 @@ def db_restore(temp_appliance_extended_db):
+    rand_filename = f"/tmp/db.backup_{fauxfactory.gen_alphanumeric()}"
     db_storage_ssh.get_file("{}/db.backup_{}".format(
         conf.cfme_data.bottlenecks.backup_path, ver), rand_filename)
     app.ssh_client.put_file(rand_filename, "/tmp/evm_db.backup")
```
We are missing this to convert to fstring.

mshriver replied:
There's no variable being formatted here?

mshriver replied:
Oh I see, you mean on the line above.

mshriver replied:
Its part of their intended design, it won't always convert.

`note: pyupgrade is intentionally timid and will not create an f-string if it would make the expression longer or if the substitution parameters are anything but simple names or dotted names (as this can decrease readability).`

mshriver commented on cfme/tests/test_replication.py:
```diff
@@ -33,7 +33,7 @@ def create_vm(provider, vm_name):
 
 def are_dicts_same(dict1, dict2):
-    logger.info("Comparing two dictionaries dict1: {%s}\n dict2:{%s}" % (dict1, dict2))
+    logger.info(f"Comparing two dictionaries dict1: {{{dict1}}}\n dict2:{{{dict2}}}")
```
This needs updating, these shouldn't be triple braced.

digitronik replied:
@mshriver done

dajoRH commented:
I detected some fixture changes in commit dcb6aeef97dc2b29d30889773edf2777f421e250

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`candu_db_restore`** is used in the following files:
- **cfme/tests/candu/test_graph_groupbytag.py**
    - *test_tagwise*

The *global* fixture **`depot_machine_ipv4_and_ipv6`** is used in the following files:
- **cfme/tests/configure/test_log_depot_operation.py**
    - *test_log_collection_over_ipv6*

The *global* fixture **`import_dialog`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
- **cfme/tests/automate/test_service_dialog.py**
    - *custom_button*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_request_details_page_tagcontrol_field*
    - *test_save_dynamic_multi_drop_down_dialog*
- **cfme/tests/services/test_dynamicdd_dialogelement.py**
    - *test_dynamic_submit_cancel_button_service*
- **cfme/tests/services/test_myservice.py**
    - *test_load_service_with_button*
- **cfme/tests/services/test_rest_services.py**
- **cfme/tests/services/test_service_customer_bz.py**
    - *test_edit_import_dialog*

The *local* fixture **`provider_credentials`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**

The *local* fixture **`default_alerts`** is used in the following files:
- **cfme/tests/control/test_default_alerts.py**
    - *test_default_alerts*

The *local* fixture **`vm_ownership`** is used in the following files:
- **cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py**
    - *resource_usage*
    - *chargeback_report_custom*

The *local* fixture **`db_restore`** is used in the following files:
- **cfme/tests/optimize/test_bottlenecks.py**
    - *test_bottlenecks_report_event_groups*
    - *test_bottlenecks_report_show_host_events*
    - *test_bottlenecks_report_time_zone*
    - *test_bottlenecks_summary_event_groups*
    - *test_bottlenecks_summary_show_host_events*
    - *test_bottlenecks_summary_time_zone*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:
